### PR TITLE
Gmarchetti/pass gateway ip explicitly

### PR DESCRIPTION
**Pull request by galenmarchetti**

### CODE REVIEW

File: commons/networks/public_ip_tracker.go [hunk 1]
```diff
@@ -21,16 +21,10 @@ func NewFreeIpAddrTracker(subnetMask string) (ipAddrTracker *FreeIpAddrTracker,
...
+	// remove the zeroth IP - it's only for marking subnet addresses.
 	_, err = ipAddrTracker.GetFreeIpAddr()
 	if err != nil {
-		return nil, stacktrace.Propagate(err, "Failed to get gatewayIP for network %v.", ipv4Net)
-	}
-	// HACK: remove the first IP - by default Docker uses this as the gateway.
-	_, err = ipAddrTracker.GetFreeIpAddr()
-	if err != nil {
-		return nil, stacktrace.Propagate(err, "Failed to get gatewayIP for network %v.", ipv4Net)
+		return nil, stacktrace.Propagate(err, "Failed to get remove zeroth IP.")
```
Comment (mieubrisse) [inline commons/networks/public_ip_tracker.go]: Think this might be a little typo

File: initializer/test_suite_runner.go [hunk 2]
```diff
@@ -135,6 +129,16 @@ func (runner TestSuiteRunner) RunTests(testNamesToRun []string) (map[string]Test
...
 		}
+		gatewayIp, err := publicIpProvider.GetFreeIpAddr()
+		if err != nil {
+			testResults[testName] = logTestResult(testName, err, false)
+			continue
+		}
+		// TODO TODO TODO Support creating one network per testnet
+		_, err = dockerManager.CreateNetwork(DEFAULT_SUBNET_MASK, gatewayIp)
+		if err != nil {
+			return nil, stacktrace.Propagate(err, "Error in creating docker subnet for testnet.")
```
Comment (mieubrisse) [inline initializer/test_suite_runner.go:140]: this should be:
```
			testResults[testName] = logTestResult(testName, err, false)
			continue
```
right?

Comment (mieubrisse) [reply initializer/test_suite_runner.go:140]: Actually don't worry about this - I'm getting it in my next PR

File: commons/networks/public_ip_tracker.go (hunk 1 above)
Comment (galenmarchetti) [reply commons/networks/public_ip_tracker.go]: done

**Review by mieubrisse — APPROVED** [review]

**Pull request merged**